### PR TITLE
ESC-250 update remove logic, enable subsidyuseageid

### PR DESCRIPTION
**Pull request by sam-miele**

### CODE REVIEW

File: app/uk/gov/hmrc/eusubsidycompliancestub/models/NonHmrcSubsidy.scala [hunk 1]
```diff
@@ -22,14 +22,14 @@ import play.api.libs.json.{Json, OFormat}
...
 case class NonHmrcSubsidy(
-  subsidyUsageTransactionId: Option[SubsidyRef], // TODO if we can't get SCP09 sorted we'll need to uppercase ID in the format
-  allocationDate: LocalDate,
-  submissionDate: LocalDate,
-  publicAuthority: Option[String], // this shouldn't be optional, is required in create API but not retrieve
-  traderReference: Option[TraderRef],
-  nonHMRCSubsidyAmtEUR: SubsidyAmount,
-  businessEntityIdentifier: Option[EORI],
-  amendmentType: Option[EisSubsidyAmendmentType] = Option.empty // this only used for create
+                           subsidyUsageTransactionID: Option[SubsidyRef], // TODO if we can't get SCP09 sorted we'll need to uppercase ID in the format
```
Comment (priyankadhiman3006) [inline app/uk/gov/hmrc/eusubsidycompliancestub/models/NonHmrcSubsidy.scala:25]: case class formatting

**Comment by priyankadhiman3006** [comment]
I think the jenkins build has failed

File: app/uk/gov/hmrc/eusubsidycompliancestub/services/Store.scala [hunk 2]
```diff
@@ -171,7 +173,7 @@ object Store {
         for {
           amendData <- amendList
           currentData <- currentList
-          bool = amendData.subsidyUsageTransactionId.get == currentData.subsidyUsageTransactionId.get
+          bool = amendData.subsidyUsageTransactionID.get == currentData.subsidyUsageTransactionID.get
```
Comment (carwynellis) [inline app/uk/gov/hmrc/eusubsidycompliancestub/services/Store.scala:176]: You can compare equality of `Option`s so the `.get` calls aren't needed.

Also `bool` could be renamed to something meaningful e.g. `hasChanges`.

**Review by carwynellis — APPROVED** [review]
Couple of minor comments you might want to address otherwise LGTM.

File: app/uk/gov/hmrc/eusubsidycompliancestub/services/Store.scala (hunk 2 above)
Comment (sam-miele) [reply app/uk/gov/hmrc/eusubsidycompliancestub/services/Store.scala:176]: Good idea. Changes made.

**Pull request merged**